hardware.h

```diff
@@ -40,7 +40,7 @@ struct MainMemory {
 	unsigned int freeFrames;
 	unsigned short nextFreeFrame;
 	struct PageTable* frames[65536];
-	int lru[65536];
+	u_int16_t lru[65536];
```

Comment:
The advantage of using this is that later on, you don't have to & with 0x0000 each time, right?

Reply:
Here, I chose `u_int16_t` in place of `int/unsigned int` because we want precise 16 bits irrespective of the system the code is running on.

https://stackoverflow.com/questions/25294594/when-to-use-uint16-t-vs-int-and-when-to-cast-type
https://stackoverflow.com/questions/17693445/what-is-the-difference-between-uint16-t-and-unsigned-short-int-incase-of-64-bit

Reply:
In this case we'll have to change everything to precise u_int values. That issue will arise in other data types also

Reply:
Not necessarily because for others we have already taken a bigger data type. Obviously if we can change it would be better